src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DeleteOperation.java

```diff
@@ -107,20 +124,77 @@ private QueryOuterClass.Query buildDeleteQuery() {
    * @param queryExecutor
    * @param query
    * @param doc
-   * @return Uni<Boolean> `true` if deleted successfully, else `false`
+   * @param retryAttempt
+   * @return Uni<Boolean> `true` if deleted successfully, else `false` if data changed and no longer
+   *     match the conditions and throws JsonApiException if LWT failure.
    */
-  private static Uni<Boolean> deleteDocument(
-      QueryExecutor queryExecutor, QueryOuterClass.Query query, ReadDocument doc) {
-    query = bindDeleteQuery(query, doc);
-    return queryExecutor
-        .executeWrite(query)
+  private Uni<Boolean> deleteDocument(
+      QueryExecutor queryExecutor,
+      QueryOuterClass.Query query,
+      ReadDocument doc,
+      AtomicInteger retryAttempt)
+      throws JsonApiException {
```

Comment:
this `throws` is suspicious and should not be there for sure

Reply:
Make sense. Removed it.